pkg/istio/clusterrbacconfig/clusterrbacconfig.go

```diff
@@ -0,0 +1,145 @@
+package clusterrbacconfig
+
+import (
+	"errors"
+	"log"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"istio.io/api/rbac/v1alpha1"
+	"istio.io/istio/pilot/pkg/config/kube/crd"
+	"istio.io/istio/pilot/pkg/model"
+)
+
+const authzEnabledAnnotation = "authz.istio.io/enable"
+
+type ClusterRbacConfigMgr struct {
+	client    *crd.Client
+	store     model.ConfigStoreCache
+	dnsSuffix string
+}
+
+// NewClusterRbacConfigMgr initializes the ClusterRbacConfigMgr object
+func NewClusterRbacConfigMgr(client *crd.Client, store model.ConfigStoreCache, dnsSuffix string) *ClusterRbacConfigMgr {
+	return &ClusterRbacConfigMgr{
+		client:    client,
+		store:     store,
+		dnsSuffix: dnsSuffix,
+	}
+}
+
+// addService will add a service to the ClusterRbacConfig object
+func (crcMgr *ClusterRbacConfigMgr) addService(service *v1.Service, clusterRbacConfig *v1alpha1.RbacConfig) bool {
+	dns := service.Name + "." + service.Namespace + "." + crcMgr.dnsSuffix
+	for _, svc := range clusterRbacConfig.Inclusion.Services {
+		if svc == dns {
+			return false
+		}
+	}
+	clusterRbacConfig.Inclusion.Services = append(clusterRbacConfig.Inclusion.Services, dns)
+	return true
+}
+
+// deleteService will delete a service from the ClusterRbacConfig object
+func (crcMgr *ClusterRbacConfigMgr) deleteService(service *v1.Service, clusterRbacConfig *v1alpha1.RbacConfig) bool {
+	var indexToRemove = -1
+	dns := service.Name + "." + service.Namespace + "." + crcMgr.dnsSuffix
+	for i, svc := range clusterRbacConfig.Inclusion.Services {
+		if svc == dns {
+			indexToRemove = i
+			break
+		}
+	}
+
+	if indexToRemove == -1 {
+		return false
+	}
+
+	clusterRbacConfig.Inclusion.Services = remove(clusterRbacConfig.Inclusion.Services, indexToRemove)
+	return true
+}
+
+// createClusterRbacConfig creates the ClusterRbacConfig object
+func (crcMgr *ClusterRbacConfigMgr) createClusterRbacConfig(service *v1.Service) model.Config {
+	return model.Config{
+		ConfigMeta: model.ConfigMeta{
+			Type:    model.ClusterRbacConfig.Type,
+			Name:    "default",
+			Group:   model.ClusterRbacConfig.Group + model.IstioAPIGroupDomain,
+			Version: model.ClusterRbacConfig.Version,
+		},
+		Spec: &v1alpha1.RbacConfig{
+			Mode: v1alpha1.RbacConfig_ON_WITH_INCLUSION,
+			Inclusion: &v1alpha1.RbacConfig_Target{
+				Services: []string{service.Name + "." + service.Namespace + "." + crcMgr.dnsSuffix},
```

Comment:
do we need the entire dnsSuffix? if  `<service.ns.svc>` works, could we skip requiring dnsSuffix completely?

Reply:
As tested, looks like it did not work without the full dns suffix.